content/Products/OpenshiftMonitoring/collecting_metrics.md

```diff
@@ -0,0 +1,184 @@
+    resources:
+      requests:
+        cpu: 1m
+        memory: 15Mi
+    securityContext:
+      allowPrivilegeEscalation: false
+      capabilities:
+        drop:
+        - ALL
+    terminationMessagePolicy: FallbackToLogsOnError
+    volumeMounts:
+    - mountPath: /etc/kube-rbac-proxy
+      name: secret-kube-rbac-proxy-metric
+      readOnly: true
+    - mountPath: /etc/tls/private
+      name: secret-kube-rbac-proxy-tls
+      readOnly: true
+    - mountPath: /etc/tls/client
+      name: metrics-client-ca
+      readOnly: true
+[...]
+  - volumes:
+    # Secret created by the service CA operator.
+    - name: secret-kube-rbac-proxy-tls
+      secret:
+        name: secret-kube-rbac-proxy-tls
+    # Secret containing the kube-rbac-proxy configuration.
+    - name: secret-kube-rbac-proxy-metric
+      secret:
+        name: secret-kube-rbac-proxy-metric
+    # ConfigMap containing the CA used to verify the client certificate.
+    - name: metrics-client-ca
+      configMap:
+        name: metrics-client-ca
+```
+
+> Note: The `metrics-client-ca` ConfigMap needs to be created by your component and synced from the `kube-system/extension-apiserver-authentication` ConfigMap.
+
+Here is a Secret containing the kube-rbac-proxy's configuration (it allows only HTTP requets to the `/metrics` endpoint for the Prometheus service account):
+
+```yaml
+apiVersion: v1
+kind: Secret
+metadata:
+  name: secret-kube-rbac-proxy-metric
+  namespace: openshift-example
+stringData:
+  config.yaml: |-
+    "authorization":
+      "static":
+      - "path": "/metrics"
+        "resourceRequest": false
+        "user":
+          "name": "system:serviceaccount:openshift-monitoring:prometheus-k8s"
+        "verb": "get"
+type: Opaque
+```
+
+Example: [node-exporter](https://github.com/openshift/cluster-monitoring-operator/blob/e51a06ffdb974003d4024ade3545f5e5e6efe157/assets/node-exporter/daemonset.yaml#L65-L98) from the Cluster Monitoring operator.
+
+### Roll your own HTTPS server
+
+> You don't use `library-go` or don't want to run a `kube-rbac-proxy` sidecar.
+
+For instance if your component is built on top of `sigs.k8s.io/controller-runtime`, the library doesn't provide any facility to configure the HTTP authn/authz as of today and it's not even possible to configure TLS (refer to https://github.com/kubernetes-sigs/controller-runtime/issues/2073 for more details). Another case might be that your component isn't an operator.
+
+In such situations, you need to implement your own HTTPS server for `/metrics`. As explained before, it needs to require and verify the TLS client certificate using the root CA stored under the `client-ca-file` key of the `kube-system/extension-apiserver-authentication` ConfigMap.
+
+In practice, the server should:
+* Set TLSConfig's `ClientAuth` field to `RequireAndVerifyClientCert`.
+* Reload the root CA when the source ConfigMap is updated.
+* Reload the server's certificate and key when they are updated.
+
+Example: TBD
```

Comment:
I still need to find an example of such component :)
otherwise https://github.com/openshift/cluster-node-tuning-operator/pull/541 once it merges

Reply:
I'll probably scratch our own itch and update CMO to run a web server supporting client certificate authentication and get rid of the kube-rbac-proxy sidecar.

Reply:
https://github.com/openshift/cluster-monitoring-operator/pull/1870